source/fuzz/fuzzer_pass_construct_composites.cpp

```diff
@@ -67,8 +67,13 @@ void FuzzerPassConstructComposites::Apply() {
         // program point) and suitable for making a synonym of, associate it
         // with the id of its result type.
         TypeIdToInstructions type_id_to_available_instructions;
-        for (auto instruction : FindAvailableInstructions(
-                 function, block, inst_it, fuzzerutil::CanMakeSynonymOf)) {
+        auto available_instructions = FindAvailableInstructions(
+            function, block, inst_it,
+            [this](opt::IRContext* ir_context, opt::Instruction* inst) {
+              return fuzzerutil::CanMakeSynonymOf(
+                  ir_context, *GetTransformationContext(), inst);
+            });
```

Comment:
If we weakened this to allow any function that can be used as a component of construct composite then we *could* allow irrelevant ids.

Then in the transformation we could only add synonym facts when ids are not irrelevant.

Reply:
Fixed in 35f5730.